conftest.py

```diff
@@ -0,0 +1,9 @@
+"""
+Configure settings before running pytest
+
+In this case, we set PYTHONHASHSEED to 0 for uniformity across test runs.
+"""
+
+import os
+
+os.environ["PYTHONHASHSEED"] = "0"
```

Comment:
Sadly, I don't believe this changes the actual value used by the interpreter (the actual hash must remain constant for the entire process). So, although it satisfies the tests, it doesn't accomplish what I'm looking for.

Honestly, I'm not sure how to do this in pre-commit. It looks like the pre-commit folks [don't want to support env variables](https://github.com/pre-commit/pre-commit/issues/1157) in the hook configuration, and I don't want to introduce something that will break on Windows. (some tests don't pass on Windows right now, but I'm actively working to fix that)

Python has a command-line option to force a random seed, but no option to un-randomize it.

So ... maybe we need to resort to using a shell Python process to set the variable? 😭 
```
python -c 'import os, subprocess; subprocess.run("python -m pytest", shell=True, env={**os.environ, "PYTHONHASHSEED":"0"})'
```

Reply:
Also! Add yourself to the contributor list in `contributing.rst` and link to whatever you like. 😄

Reply:
So `conftest.py` is a special file for `pytest` that runs before any tests are run, and when I ran the `pytest` locally before and after adding this, I did see the pytest that checks the seed stop failing, for what it's worth.  It should also not be permanent, as adding things to `os.environ` only last for the duration of the Python interpreter process, if I understand correctly.  Let me know if I'm misunderstanding the issue though.

Reply:
And I'll go ahead and add myself, and the other changes when those are decided!

Reply:
I also meant to mention that this pytest hook runs locally, so using `conftest.py` to set the environment should definitely work!

Reply:
Yeah, so changing `os.environ` will trick my silly test into passing, but unfortunately it doesn't change the underlying seed used by python. Check this out:
```
$ PYTHONHASHSEED=0 python -c 'print(hash("foo"))'
-2457967226571033580
$ PYTHONHASHSEED=0 python -c 'print(hash("foo"))'
-2457967226571033580
$ python -c 'import os; os.environ["PYTHONHASHSEED"] = "0";  print(hash("foo"))'
3608991658523737785
$ python -c 'import os; os.environ["PYTHONHASHSEED"] = "0";  print(hash("foo"))'
-5911599181240111572
```
When we set the environment variable before python starts, the behavior of hash is consistent across runs. If we set it in `os.environ`, it changes across runs. (b/c by then the Python interpreter doesn't care - it's already picked its seed)

Reply:
So I think we can get around it be using this entry:

```
entry: env PYTHONHASHSEED=0 python -m pytest
```

Looks like that's the reported fix, and sets it right before calling Python and only for that specific spin of the interpreter based on my local testing.

Reply:
> entry: env PYTHONHASHSEED=0 python -m pytest

My only reservation for this solution is that it feels like it won't work on windows (unless it's cygwin python perhaps?). I could test this tomorrow. I've got other problems with windows, so it's not the biggest deal, but it's one more thing that we'll have to unwind eventually. Anyway, that's why I was proposing that horrendous `python -c` construction earlier. What do you think?

Reply:
Tested on my Windows machine just now, and getting the same result using `cmd`:

```shell
C:\Users\adelaney>python
Python 3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.environ["PYTHONHASHSEED"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\adelaney\AppData\Local\Programs\Python\Python39\lib\os.py", line 679, in __getitem__
    raise KeyError(key) from None
KeyError: 'PYTHONHASHSEED'
>>> quit()

C:\Users\adelaney>env PYTHONHASHSEED=0 python
Python 3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.environ["PYTHONHASHSEED"]
'0'
```

I'm open to either solution though, but I agree that avoiding the awful `python -c` construction would be great.

Reply:
Oh wow. I stand corrected! Great, let's go with the `env` approach then!

Reply:
Sounds like a plan!